src/main/java/ws/wamp/jawampa/WampRouter.java

```diff
@@ -676,32 +700,71 @@ private void onMessageFromRegisteredChannel(WampRouterHandler handler, WampMessa
-                    if (receiver == handler) { // Potentially skip the publisher
-                        boolean skipPublisher = true;
-                        if (pub.options != null) {
-                            JsonNode excludeMeNode = pub.options.get("exclude_me");
-                            if (excludeMeNode != null) {
-                                skipPublisher = excludeMeNode.asBoolean(true);
-                            }
+            Subscription exactSubscription = handler.realm.subscriptionsByFlags.get(SubscriptionFlags.Exact).get(pub.topic);
+            if (exactSubscription != null) {
+                publishEvent(handler, pub, publicationId, exactSubscription);
+            }
+
+            Map<String, Subscription> prefixSubscriptionMap = handler.realm.subscriptionsByFlags.get(SubscriptionFlags.Prefix);
+            for (Subscription prefixSubscription : prefixSubscriptionMap.values()) {
+                if (pub.topic.startsWith(prefixSubscription.topic)) {
+                    publishEvent(handler, pub, publicationId, prefixSubscription);
+                }
+            }
+
+            Map<String, Subscription> wildcardSubscriptionMap = handler.realm.subscriptionsByFlags.get(SubscriptionFlags.Wildcard);
+            String[] components = pub.topic.split("\\.", -1);
+            for (Subscription wildcardSubscription : wildcardSubscriptionMap.values()) {
+                boolean matched = true;
+                if (components.length == wildcardSubscription.components.length) {
+                    for (int i=0; i < components.length; i++) {
+                        if (wildcardSubscription.components[i].length() > 0
+                                && !components[i].equals(wildcardSubscription.components[i])) {
+                            matched = false;
+                            break;
                         }
-                        if (skipPublisher) continue;
                     }
-                    // Publish the event to the subscriber
-                    EventMessage ev = new EventMessage(subscription.subscriptionId, publicationId,
-                        null, pub.arguments, pub.argumentsKw);
-                    receiver.ctx.writeAndFlush(ev);
+                }else
+                    matched = false;
+
+                if (matched) {
+                    publishEvent(handler, pub, publicationId, wildcardSubscription);
                 }
             }
-            
+
             if (sendAcknowledge) {
                 PublishedMessage response = new PublishedMessage(pub.requestId, publicationId);
                 handler.ctx.writeAndFlush(response);
             }
         }
     }
+
+    private void publishEvent(WampRouterHandler publisher, PublishMessage pub, long publicationId, Subscription subscription){
+        ObjectNode details = null;
+        if (subscription.flags != SubscriptionFlags.Exact) {
+            details = new ObjectMapper().createObjectNode();
```

Comment:
Should use the objectMapper field of the router class instead of creating a new one